core/services/repo_manager.rb

```diff
@@ -107,17 +116,48 @@ def addRepo(file)
 
   private
 
-  # Find the latest available repository version
+  # Find all available product repositories by repository_key
   # @param product [Hash] hash array with information about product
   # @param repository_key [String] key of repository
-  def find_last_repository_version(product, repository_key)
-    all_available_repo = @repos.find_all do |elem|
+  def find_available_repo(product, repository_key)
+    @repos.find_all do |elem|
       elem[1]['product'] == product['name'] &&
         elem[1]['platform'] + '^' + elem[1]['platform_version'] == repository_key
     end
+  end
+
+  # Find the latest available repository version
+  # @param product [Hash] hash array with information about product
+  # @param repository_key [String] key of repository
+  def find_last_repository_version(product, repository_key)
+    all_available_repo = find_available_repo(product, repository_key)
     repo = all_available_repo.last[1]
     all_available_repo.delete_if { |elem| elem[1]['version'].include?('debug') }
     repo = all_available_repo.last[1] unless all_available_repo.nil?
     repo
   end
+
+  def parse_sem_version(version)
+    return nil unless SemVersion.valid?(version)
```

Comment:
You should apply the same rules to enhance the version to SemVer as you are doing to the required version.